api-processor/src/main/java/com/synopsys/integration/alert/processor/api/filter/JobNotificationMapper.java

```diff
@@ -51,39 +51,82 @@ public JobNotificationMapper(ProcessingJobAccessor processingJobAccessor) {
         return new StatefulAlertPage<>(firstPage, filteredJobWrapperPageRetriever, HAS_NEXT_PAGE);
     }
 
-    private AlertPagedDetails<FilteredJobNotificationWrapper> mapPageOfJobsToNotification(List<DetailedNotificationContent> detailedContents, List<FrequencyType> frequencies, int pageNumber, int pageSize) {
+    private AlertPagedDetails<FilteredJobNotificationWrapper> mapPageOfJobsToNotification(
+        List<DetailedNotificationContent> detailedContents,
+        List<FrequencyType> frequencies,
+        int pageNumber,
+        int pageSize
+    ) {
         if (detailedContents.isEmpty()) {
             return new AlertPagedDetails<>(1, pageNumber, pageSize, List.of());
         }
 
         Map<FilteredDistributionJobResponseModel, List<NotificationContentWrapper>> groupedFilterableNotifications = new HashMap<>();
 
-        FilteredDistributionJobRequestModel filteredDistributionJobRequestModel = createRequestModelFromNotifications(detailedContents, frequencies);
-
-        AlertPagedDetails<FilteredDistributionJobResponseModel> jobs = processingJobAccessor.getMatchingEnabledJobsByFilteredNotifications(filteredDistributionJobRequestModel, pageNumber, pageSize);
-        for (DetailedNotificationContent detailedNotificationContent : detailedContents) {
-            for (FilteredDistributionJobResponseModel filteredDistributionJobResponseModel : jobs.getModels()) {
-                if (JobNotificationFilterUtils.doesNotificationApplyToJob(filteredDistributionJobResponseModel, detailedNotificationContent)) {
-                    List<NotificationContentWrapper> applicableNotifications = groupedFilterableNotifications.computeIfAbsent(filteredDistributionJobResponseModel, ignoredKey -> new LinkedList<>());
-                    applicableNotifications.add(detailedNotificationContent.getNotificationContentWrapper());
+        int totalPages = 0;
+        List<FilteredDistributionJobRequestModel> filteredDistributionJobRequestModels = createRequestModelFromNotifications(detailedContents, frequencies);
+        for (FilteredDistributionJobRequestModel filteredDistributionJobRequestModel : filteredDistributionJobRequestModels) {
+            AlertPagedDetails<FilteredDistributionJobResponseModel> jobs = processingJobAccessor.getMatchingEnabledJobsByFilteredNotifications(
+                filteredDistributionJobRequestModel,
+                pageNumber,
+                pageSize
+            );
+            totalPages += jobs.getTotalPages();
+            for (DetailedNotificationContent detailedNotificationContent : detailedContents) {
+                for (FilteredDistributionJobResponseModel filteredDistributionJobResponseModel : jobs.getModels()) {
+
+                    if (JobNotificationFilterUtils.doesNotificationApplyToJob(filteredDistributionJobResponseModel, detailedNotificationContent) &&
+                        detailedNotificationContent.getProviderConfigId() == filteredDistributionJobRequestModel.getProviderConfigId()) {
```

Comment:
New filter here to prevent unrelated notifications from being inserted into the object